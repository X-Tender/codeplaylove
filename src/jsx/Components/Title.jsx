import React from 'react';
import PropTypes from 'prop-types';

const MaskedSVG = ({ text, type, color }) => (
	<svg className="title__svg" height="100%" width="100%">
		<rect
			className="knockout-text-bg"
			fill={color}
			fillOpacity="0.4"
			height="100%"
			mask={`url(#knockout-text-${type})`}
			width="100%"
			x="0"
			y="0"
		/>

		<mask className="title__mask" id={`knockout-text-${type}`}>
			<rect fill="#FFF" height="100%" width="100%" x="0" y="0" />
			<text
				alignmentBaseline="central"
				dominantBaseline="central"
				fill="#000"
				textAnchor="middle"
				x="50%"
				y="50%"
			>
				{text}
			</text>
		</mask>
	</svg>
);

MaskedSVG.propTypes = {
	text: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
	color: PropTypes.string.isRequired,
};

const Code = () => (
	<span className="title__text title__text--code">
		<MaskedSVG color="#8ac926" text="code" type="code" />
	</span>
);
const Play = () => (
	<span className="title__text title__text--play">
		<MaskedSVG color="#2e84d1" text="PLAY" type="play" />
	</span>
);
const Love = () => (
	<span className="title__text title__text--love">
		<MaskedSVG color="#fc5053" text="love" type="love" />
	</span>
);

const Title = () => (
	<div className="title">
		<h1 className="title__head">
			<Code />
			<Play />
			<Love />
		</h1>
	</div>
);

export default Title;
