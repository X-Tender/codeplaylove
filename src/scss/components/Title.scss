.title {
	position: absolute;
	left: 0;
	top: 0;
	size: 100%;

	&__head {
		width: 100%;
		height: 100%;
	}

	&__text {
		position: relative;
		font-weight: 400;
		text-align: center;
		font-size: 20vh;
		font-family: 'Raleway', sans-serif;
		text-transform: uppercase;
		font-weight: 900;
		height: 33.33333%;
		display: block;

		@include sm {
			font-size: 25vh;
		}
	}

	&__svg {
	}

	&__mask {
	}
}
