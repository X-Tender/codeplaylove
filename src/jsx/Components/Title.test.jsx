import React from 'react';
import { shallow } from 'enzyme';
import Title from './Title';

describe('Title', () => {
	const comp = shallow(<Title />);

	it('renders correctly', () => {
		expect(comp).toMatchSnapshot();
	});
});
