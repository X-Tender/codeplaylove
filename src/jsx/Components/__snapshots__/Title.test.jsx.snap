// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders correctly 1`] = `
<div
  className="title"
>
  <h1
    className="title__head"
  >
    <TitleSegment
      color="#8ac926"
      type="code"
    >
      code
    </TitleSegment>
    <TitleSegment
      color="#2e84d1"
      type="play"
    >
      PLAY
    </TitleSegment>
    <TitleSegment
      color="#fc5053"
      type="love"
    >
      love
    </TitleSegment>
  </h1>
</div>
`;
